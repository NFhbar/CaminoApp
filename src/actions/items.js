import uuid from 'uuid';

// ADD_ITEM
export const addItem = (
  {
    description = '',
    note = '',
    amount = 0,
    createdAt = 0
  } = {}
) => ({
  type: 'ADD_ITEM',
  item: {
    id: uuid(),
    description,
    note,
    amount,
    createdAt
  }
});

// REMOVE_ITEM
export const removeItem = ({ id } = {}) => ({
  type: 'REMOVE_ITEM',
  id
});

// EDIT_ITEM
export const editItem = (id, updates) => ({
  type: 'EDIT_ITEM',
  id,
  updates
});
