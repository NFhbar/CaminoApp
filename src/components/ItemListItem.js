import React from 'react';
import {Link} from 'react-router-dom';
import numeral from 'numeral';

const ItemListItem = ({id, description, amount, createdAt}) => (
  <div>
    <Link to={`/edit/${id}`}>
      <h3>{description}</h3>
    </Link>
    <p>{numeral(amount/100).format('$0,0.00')} - {createdAt}</p>
  </div>
);

export default ItemListItem;
