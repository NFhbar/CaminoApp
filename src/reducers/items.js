// Items Reducer

const itemsReducerDefaultState = [];

export default (state = itemsReducerDefaultState, action) => {
  switch (action.type) {
    case 'ADD_ITEM':
      return [
        ...state,
        action.item
      ];
    case 'REMOVE_ITEM':
      return state.filter(({ id }) => id !== action.id);
    case 'EDIT_ITEM':
      return state.map((item) => {
        if(item.id === action.id){
          return {
            ...item,
            ...action.updates
          };
        } else {
          return item;
        };
      });
    default:
      return state;
  }
};
