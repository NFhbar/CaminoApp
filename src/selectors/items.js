import moment from 'moment';

// Get visible items

export default (items, { text, sortBy, startDate, endDate}) => {
  return items.filter((item) => {
    const createdAtMoment = moment(item.createdAt);
    const startDateMatch = startDate ? startDate.isSameOrBefore(createdAtMoment,'day'): true ;
    const endDateMatch = endDate ? endDate.isSameOrAfter(createdAtMoment, 'day'): true;
    const textMatch = item.description.toLowerCase().includes(text.toLowerCase());

    return startDateMatch && endDateMatch && textMatch;
  }).sort((a, b) => {
    if(sortBy === 'date') {
      return a.createdAt < b.createdAt ? 1 : -1;
    } else if (sortBy === 'amount') {
      return a.amount < b.amount ? 1: -1;
    }
  });
};
