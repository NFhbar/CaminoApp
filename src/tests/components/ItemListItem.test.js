import React from 'react';
import { shallow } from 'enzyme';
import ItemListItem  from '../../components/ItemListItem';
import items from '../fixtures/items';

test('should render ItemListItem correctly', () => {
  const wrapper = shallow(<ItemListItem {...items[0]} />);
  expect(wrapper).toMatchSnapshot();
});
