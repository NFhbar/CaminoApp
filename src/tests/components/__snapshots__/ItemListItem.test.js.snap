// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ItemListItem correctly 1`] = `
<div>
  <Link
    replace={false}
    to="/edit/1"
  >
    <h3>
      Pant
    </h3>
  </Link>
  <p>
    500
     - 
    0
  </p>
</div>
`;
